lib/vector-core/src/event/metric/data.rs

```diff
@@ -83,6 +90,28 @@ impl MetricData {
                 (Some(t), None) | (None, Some(t)) => Some(t),
                 (Some(t1), Some(t2)) => Some(t1.max(t2)),
             };
+
+            let delta_t = self
+                .timestamp
+                .and_then(|ts| {
+                    other
+                        .timestamp
+                        .map(|other_ts| (ts.timestamp_millis() - other_ts.timestamp_millis().abs()))
```

Comment:
I think the `.abs` is in the wrong place here. Shouldn't it go outside of the subtraction? Also, there is now `.abs_diff`:
```suggestion
                        .map(|other_ts| ts.timestamp_millis().abs_diff(other_ts.timestamp_millis()))
```